dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasableResourceInterface.java

```diff
@@ -67,9 +67,9 @@ public interface AliasableResourceInterface<T extends Aliasable> {
      * @param user user authenticated to issue a DOI for the workflow
      * @param blockAliasesWithZenodoFormat block creation of an alias with a particular format
      */
-    static void checkAliases(Set<String>  aliases, User user, boolean blockAliasesWithZenodoFormat) {
+    static void checkAliases(Set<String>  aliases, Optional<User> user, boolean blockAliasesWithZenodoFormat) {
         // Admins and curators do not have restrictions on alias format
-        if (user.isCurator() || user.getIsAdmin()) {
+        if (user.isPresent() && user.get().isCurator() || user.get().getIsAdmin()) {
```

Comment:
This will throw if the user isn't present, needs parentheses:
```
user.isPresent() && (user.get().isCurator() || user.get().getIsAdmin())
```
A "safer" more "Optional-oriented" version is something like:
```
user.map(user -> user.isCurator() || user.getIsAdmin()).orElse(false)
```
But it's harder to read, IMHO.

Reply:
Good catch, that was sloppy of me.

Reply:
Went for readable.